crates/sui-prover/tests/inputs/quantifiers/complex_usage.ok.move

```diff
@@ -0,0 +1,30 @@
+#[allow(unused)]
+module 0x42::quantifiers_complex_usage;
+
+#[spec_only]
+use prover::prover::{exists, ensures, invariant};
+use prover::vector_iter::map;
+
+fun invariant_expression(j: u64, i: u64, u_j: u8, v_j: u8): bool {
+    j < i && u_j > v_j
+}
+
+fun vec_leq(u: vector<u8>, v: vector<u8>): bool {
+    if (u.length() > v.length()) {
+        return false;
+    };
+    let mut i = 0;
+    invariant!(|| ensures(i <= u.length() && !exists!<u64>(|j| invariant_expression(*j, i, u[*j], v[*j]))));
```

Comment:
this would be nice, but I don't think it's easy to support now. I was thinking of something simpler
```
fun invariant_expression(j: &u64, i: u64, u: &vector<u8>, v: &vector<u8>): bool {
    if (j < u.length() && j < v.length()) {
        j < i && u[*j] > v[*j]
    } else {
        false
    }
}
...
invariant!(|| ensures(i <= u.length() && !exists!<u64>(|j| invariant_expression(*j, i, &u, &v))));
```